src/query/pipeline/sources/src/processors/sources/input_formats/input_pipeline.rs

```diff
@@ -203,13 +203,26 @@ pub trait InputFormatPipe: Sized + Send + 'static {
                 let splits = splits.to_owned().clone();
                 tokio::spawn(async move {
                     let mut futs = FuturesUnordered::new();
-                    for s in &splits {
-                        let fut = Self::read_split(ctx_clone2.clone(), s);
+                    for s in splits.into_iter() {
+                        let fut =
+                            CatchUnwindFuture::create(Self::read_split(ctx_clone2.clone(), s));
                         futs.push(fut);
                     }
                     while let Some(row_batch) = futs.next().await {
-                        if data_tx2.send(row_batch.unwrap()).await.is_err() {
-                            info!("execute_copy_aligned fail to send row_batch")
+                        match row_batch {
+                            Ok(row_batch) => {
+                                let is_err = row_batch.is_err();
+                                if data_tx2.send(row_batch).await.is_err() {
+                                    break;
+                                }
+                                if is_err {
+                                    break;
+                                }
+                            }
+                            Err(cause) => {
+                                data_tx2.send(Err(cause)).await.ok();
+                                break;
+                            }
```

Comment:
```suggestion
match row_batch {
    Ok(row_batch) => {
        if let Err(cause) = data_tx2.send(row_batch).await {
            break;
        }
    }
    Err(cause) => {
        if let Err(send_error) = data_tx2.send(Err(cause)).await {
            break;
        }
    }
}
```

Reply:
this not help with `on_error=continue`,  because the receiver side will abort the query,  should not try to read/send data any more.

as I said, there are many other kinds of error to handle (not only read row) to better support `on_error=continue` .
compare with lost data silently,  I prefer fail for  errors happend here.

Reply:
cc @ariesdevil

Reply:
Why not:
```rust
if row_batch.ok() {
	if data_tx2.send(row_batch).await.is_err() {
	    break;
	}
} else {
    break;
}
```

Reply:
why?

Reply:
done @ariesdevil